### PR TITLE
Issue #13345: Enabled SeverityMatch example tests

### PR DESCRIPTION
**Pull request by SteLeo1602**
Issue: #13345

Enabled the example tests for SeverityMatch filter.

### CODE REVIEW

**Comment by SteLeo1602** [comment]
GitHub, generate website.

**Comment by SteLeo1602** [comment]
GitHub, generate website

File: src/xdocs/filters/severitymatchfilter.xml [hunk 1]
```diff
@@ -67,6 +67,16 @@
...
 &lt;/module&gt;
         </source>
+        <p id="Example1-code">
+            Example code:
+        </p>
+        <source>
+public class Example1 {
+  public void method1(int V1){} // ok, ParameterNameCheck's severity is info
+
+  public void Method2(){} // violation, 'must match pattern'
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/filters/severitymatchfilter.xml]: Do I have to change it one by one? If so, how do I make it pass the CI?

[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs/filters/severitymatchfilter.xml]: Strange, we just changed wording of explanations, can you check if `'` in reasons is root of problems?

[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs/filters/severitymatchfilter.xml]: https://regex101.com/r/yNND7u/1

[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs/filters/severitymatchfilter.xml]: we should match to VIOLATION_WITH_EXPLANATION_PATTERN

[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs/filters/severitymatchfilter.xml]: I pushed extra commit to let you see more clearly what was a problem

[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/filters/severitymatchfilter.xml]: Ok, squashed the commit

**Review by romani — APPROVED** [review]
Ok to merge.

Removed violation comments are good as they are at wrong place, they were never active, with fix become active - required to be removed

**Comment by SteLeo1602** [comment]
@rnveach I wanted to remind you to review my changes here. The last feedback was over a week ago and I think I should notify you by now.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ public final class InlineConfigParser {
 
     /** A pattern to find the string: "// violation, explanation". */
     private static final Pattern VIOLATION_WITH_EXPLANATION_PATTERN = Pattern
-            .compile(".*//\\s*violation,\\s.+\\s(?:['\"](.*)['\"])?$");
+            .compile(".*//\\s*violation,\\s.+\\s(?:.*)?$");
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: Why was this changed? No other PRs so far have had an issue.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java [hunk 3]
```diff
@@ -630,11 +630,8 @@ else if (violationBelowWithExplanationMatcher.matches()) {
             inputConfigBuilder.addViolation(violationLineNum, violationMessage);
         }
         else if (violationWithExplanationMatcher.matches()) {
-            final String violationMessage = violationWithExplanationMatcher.group(1);
             final int violationLineNum = lineNo + 1;
-            checkWhetherViolationSpecified(specifyViolationMessage, violationMessage,
-                    violationLineNum);
-            inputConfigBuilder.addViolation(violationLineNum, violationMessage);
+            inputConfigBuilder.addViolation(violationLineNum, null);
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:634]: Same for this code.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java [hunk 4]
```diff
@@ -12,15 +12,15 @@ public boolean equals(String o) {
             return true;
         }
     }
-    public static class TestClass2 { // violation, no correct `equals`
+    public static class TestClass2 {
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: This also.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/severitymatchfilter/Example1.java [hunk 5]
```diff
@@ -16,8 +16,8 @@
 
 // xdoc section -- start
 public class Example1 {
-  public void method1(int V1){} // ok, ParameterNameCheck's severity is info
+  public void method1(int V1){} // ok, ParameterName's severity is info
 
-  public void Method2(){} // violation, 'must match pattern'
+  public void Method2(){} // violation, MethodName severity is default it is error
```
Comment (rnveach) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/filters/severitymatchfilter/Example1.java:21]: `MethodName severity is default it is error` => `MethodName's severity is defaulted to error`

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java (hunk 2 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: Nobody noticed defect in this code before until this migration.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java (hunk 4 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: This location of comment is wrong, it was ignored before. With this update it becomes actionable and can not stay at this line

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java (hunk 2 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: What was the defect?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java (hunk 4 above)
Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: Why was this missed? Is this connected to the regexp change?

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java (hunk 2 above)
Comment (SteLeo1602) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: Before it assumed that there should be quotation marks around the explanation just like it would be around violation's message

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java (hunk 4 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: Regexp was bad, all such lines were ignored.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java (hunk 2 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: It was copy paste mistake, fortunately no much usage of this violation style in use.
This PR is first actual usage of this style.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java (hunk 4 above)
Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: If this is an issue we ignored such lines, won't we ignore more lines that don't match the regexp? What about `violation` with no comma? Or with comma but nothing afterwards?

Why aren't we finding all violation lines, and then checking if they match a message and if they don't then fail with an unrecognized message.

Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: It also sounds like this should be split into 2 commits. 1 on bad regexp, other on severity match which PR is tied to.

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: @SteLeo1602 , ok, this update becomes confusing for others.

Let's create issue on fact that trailing comments without message but with some reason wording is not working. We will fix only it in scope of this issue, separate PR.

And after merging of such update, we will enable this Example.

Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: I don't feel any confusion. It seems like there is more going on here and a more deep investigation needs to be done.

We completely missed a `// violation` comment, and that can happen again regardless of what we update regular expression to.

Also this PR was definitely missing explanations on what its doing and why.
https://github.com/checkstyle/checkstyle/pull/15021#discussion_r1670913625
Never got an explanation on this.

Just cause this is tied to an existing issue, doesn't mean we should make a whole bunch of what appears to be unrelated updates without an explanation. If we find bugs, either make them their own PR or tie them to their own issue separate from approved work.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java (hunk 2 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: `".*//\\s*violation,\\s.+\\s(?:.*)?$"` => `".*//\\s*violation,\\s+(?:.*)?$"`

The extra `.*\\s` doesn't really make sense.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java (hunk 4 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java:15]: @SteLeo1602
new issue is created.
https://github.com/checkstyle/checkstyle/issues/15256

please fix only it in separate PR.

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java (hunk 2 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:101]: moved to https://github.com/checkstyle/checkstyle/issues/15256

**Comment by rnveach** [comment]
Closing this. Please create a new PR with other admins as the reviewers.
https://github.com/checkstyle/checkstyle/issues/15643#issuecomment-2336777333
